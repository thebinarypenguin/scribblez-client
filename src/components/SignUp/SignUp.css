#SignUp .dialog {
  font-size: 1.5rem;
  text-align: center;
  padding-top: 4rem;
  padding-bottom: 4rem;
}

#SignUp h2 {
  margin-bottom: 2rem;
}


#SignUp input {
  display: block;
  margin: auto;
  width: 20rem;
  padding: 3px 6px;
  margin-bottom: 1rem;
}


#SignUp button {
  padding: 3px 6px;
}
