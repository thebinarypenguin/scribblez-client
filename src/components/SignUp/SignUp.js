import React from 'react';
import AuthService from '../../services/AuthService';
import UserService from '../../services/UserService';
import UserContext from '../../contexts/UserContext';
import TopBar from '../../components/TopBar/TopBar';
import Utils from '../../services/Utils';


import './SignUp.css';

class SignUp extends React.Component {

  static contextType = UserContext;

  constructor(props) {
    super(props);

    this.state = {
      error: null,
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.generateLinks = this.generateLinks.bind(this);
  }

  generateLinks() {

    let links = [];

    if (Utils.isTokenActive(window.localStorage.getItem('token'))) {

      links = [
        { name: 'Feed',     target: '/feed'     },
        { name: 'Notes',    target: '/notes'    },
        { name: 'Sign Out', target: '/sign-out' },
      ];

    } else {

      links = [
        { name: 'Feed',    target: '/feed'    },
        { name: 'Sign Up', target: '/sign-up' },
        { name: 'Sign In', target: '/sign-in' },
      ];
    }

    return links;
  }

  handleSubmit(ev) {

    ev.preventDefault();

    const payload = {
      real_name     : ev.target['full-name'].value,
      email_address : ev.target['email_address'].value,
      username      : ev.target['username'].value,
      password      : ev.target['password'].value,
    };

    // console.log('IOU form validation');

    return UserService
      .createUser(payload)
      .then(() => {
        return AuthService.login(payload.username, payload.password);
      })
      .then((resp) => {
        // console.log(resp)
        this.context.logIn(resp.token);
        this.props.history.push('/notes');
      })
      .catch((err) => {
        this.setState({ error: err.message });
      });
  }

  render() {

    return (
      <div id="SignUp">

        <TopBar links={this.generateLinks()} />

        <div className="dialog">

          <h2>Sign Up</h2>

          <form autoComplete="off" onSubmit={this.handleSubmit}>

            <input type="text" id="full-name" placeholder="Name" />
            <input type="text" id="email_address" placeholder="Email Address" />
            <input type="text" id="username" placeholder="Username" />
            <input type="password" id="password" placeholder="Password" />
            <input type="password" id="password-confirmation" placeholder="Confirm Password" />
            <button type="submit">Sign Up</button>

          </form>

      </div>

    </div>
    );
  }
}

export default SignUp;
