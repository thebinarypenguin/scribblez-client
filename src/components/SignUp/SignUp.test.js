import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import SignUp from './SignUp';

it('renders without crashing', () => {

  const container = document.createElement('div');

  const subject = (
    <MemoryRouter>
      <SignUp />
    </MemoryRouter>
  );

  ReactDOM.render(subject, container);
  ReactDOM.unmountComponentAtNode(container);
});

it('renders the UI as expected', () => {

  const subject = (
    <MemoryRouter>
      <SignUp />
    </MemoryRouter>
  );

  const tree = renderer.create(subject).toJSON();
  expect(tree).toMatchSnapshot();
});
