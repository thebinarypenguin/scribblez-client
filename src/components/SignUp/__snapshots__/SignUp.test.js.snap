// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  id="SignUp"
>
  <div
    id="TopBar"
  >
    <div
      className="Container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1>
          Scribblez
        </h1>
      </a>
      <ul>
        <li>
          <a
            href="/feed"
            onClick={[Function]}
          >
            Feed
          </a>
        </li>
        <li>
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </li>
        <li>
          <a
            href="/sign-in"
            onClick={[Function]}
          >
            Sign In
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="dialog"
  >
    <h2>
      Sign Up
    </h2>
    <form
      autoComplete="off"
      onSubmit={[Function]}
    >
      <input
        id="full-name"
        placeholder="Name"
        type="text"
      />
      <input
        id="email_address"
        placeholder="Email Address"
        type="text"
      />
      <input
        id="username"
        placeholder="Username"
        type="text"
      />
      <input
        id="password"
        placeholder="Password"
        type="password"
      />
      <input
        id="password-confirmation"
        placeholder="Confirm Password"
        type="password"
      />
      <button
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;
